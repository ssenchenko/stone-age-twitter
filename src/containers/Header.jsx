import React from 'react';
import ReactRouterPropTypes from 'react-router-prop-types';
import styled from 'styled-components';
import { withRouter } from 'react-router-dom';

import Logo from '../components/Logo';
import NotLoggedNavBar from './NotLoggedNavBar';
import ErrorBoundary from './ErrorBoundary';
import * as routes from '../settings/routes';

const HeaderStyled = styled.header`
  position: sticky;

  display: flex;
  flex-flow: row wrap;
  justify-content: space-around;
  align-content: space-around;

  font-weight: 600;
  border-block-end: thin solid lightgreen;
`;

const LogoWrapper = styled.div`
  order: 1{/*1st in the flexbox*/};
  flex: 1{/*flex-grow = 1*/};
`;

const PlaceHolder = styled.div`
  order: 2;
  flex: 2{/*flex-grow = 2*/};
`;

const NavBar = styled.div`
  order: 3;
  flex: 1{/*flex-grow = 1*/};
`;

const HeaderBase = ({ history }) => {
  const goToLogin = () => {
    history.push(routes.LOGIN);
  };

  const goToSignUp = () => {
    history.push(routes.SIGN_UP);
  };

  return (
    <HeaderStyled>
      <LogoWrapper>
        <Logo />
      </LogoWrapper>
      <PlaceHolder />
      <NavBar>
        <ErrorBoundary>
          <NotLoggedNavBar visible onLoginClick={goToLogin} onSignUpClick={goToSignUp} />
        </ErrorBoundary>
      </NavBar>
    </HeaderStyled>
  );
};

HeaderBase.propTypes = {
  history: ReactRouterPropTypes.history.isRequired,
};

const Header = withRouter(HeaderBase);

export default Header;
export { HeaderBase };
