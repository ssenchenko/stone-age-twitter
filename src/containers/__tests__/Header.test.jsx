import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { MemoryRouter } from 'react-router-dom';

import { HeaderBase } from '../Header';
import DefaultTheme from '../../settings/themes';

const historyMock = {
  push: jest.fn(),
};
const locationMock = { push: jest.fn() };

describe('HeaderBase', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={DefaultTheme}>
          <MemoryRouter>
            <HeaderBase history={historyMock} location={locationMock} />
          </MemoryRouter>
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
