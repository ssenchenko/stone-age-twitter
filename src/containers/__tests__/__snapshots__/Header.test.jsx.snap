// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBase renders correctly 1`] = `
.c1 {
  color: darkgreen;
  font-weight: bold;
  font-size: 32px;
}

.c0 {
  position: -webkit-sticky;
  position: sticky;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-content: space-around;
  -ms-flex-line-pack: space-around;
  align-content: space-around;
  font-weight: 600;
  border-block-end: thin solid lightgreen;
  margin-bottom: 10px;
}

.c2 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

<header
  className="c0"
>
  <div
    className=""
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <h1
        className="c1"
      >
        Test
      </h1>
    </a>
  </div>
  <div
    className="c2"
  />
</header>
`;
